packages/lambdas/src/sqs-to-converter.ts

```diff
@@ -59,24 +65,123 @@ function isSidechainConnector() {
   return sidechainFHIRConverterUrl ? true : false;
 }
 
-async function getSidechainConverterAPIKey(): Promise<string> {
-  if (!sidechainFHIRConverterKeysSecretName) {
-    throw new Error(`Programming error - sidechainFHIRConverterKeys is not set`);
+async function getAndUpdateSidechainConverterKeys(): Promise<string> {
+  if (!sidechainKeysTableName) {
+    throw new Error(`Programming error - SIDECHAIN_FHIR_CONVERTER_KEYS_TABLE_NAME is not set`);
   }
 
-  const secret = await getSecret(sidechainFHIRConverterKeysSecretName);
-  if (!secret) {
-    throw new Error(`Config error - sidechainFHIRConverterKeysSecret doesn't exist`);
+  const docClient = new DynamoDB.DocumentClient({
+    apiVersion: "2012-08-10",
+  });
+
+  // get all keys that haven't been revoked
+  const keysTableItems = await docClient
+    .scan({
+      TableName: sidechainKeysTableName,
+      FilterExpression: "attribute_exists(status) AND (status = :active OR status = :rateLimit)",
+      ExpressionAttributeValues: {
+        ":active": sidechainKeysStatus.active.toString(),
+        ":rateLimit": sidechainKeysStatus.rateLimit.toString(),
+      },
+    })
+    .promise();
+
+  if (!keysTableItems.Items) {
+    throw new Error(`Programming error - no keys found in table ${sidechainKeysTableName}`);
+  }
+  let activeKey = "";
+  const keysToUpdate: string[] = [];
+  for (const keyItem of keysTableItems.Items) {
+    if (!keyItem.status || !keyItem.key) {
+      throw new Error(
+        `Programming error - keys found in table ${sidechainKeysTableName} are not the expected format`
+      );
+    }
+    const rateLimitDate = keyItem.rateLimitDate ? dayjs(keyItem.rateLimitDate) : undefined;
+    const now = dayjs();
+    if (keyItem.status === sidechainKeysStatus.active.toString()) {
+      if (!activeKey) {
+        // pick the first active API key
+        activeKey = keyItem.key;
+      }
+    } else if (
+      keyItem.status === sidechainKeysStatus.rateLimit.toString() &&
+      rateLimitDate &&
+      now.isAfter(rateLimitDate) &&
+      now.month() > rateLimitDate.month()
+    ) {
+      // this key should have its rate limit reset by now
+      keysToUpdate.push(keyItem.key);
+      if (!activeKey) {
+        activeKey = keyItem.key;
+      }
+    }
   }
 
-  const keys = String(secret).split(",");
-  if (keys.length < 1) {
-    throw new Error(
-      `Config error - sidechainFHIRConverterKeysSecret needs to be a comma separated string of keys`
-    );
+  const newKeyItemPutRequests = keysToUpdate.map(key => {
+    return {
+      PutRequest: {
+        Item: {
+          TableName: sidechainKeysTableName,
+          key,
+          status: sidechainKeysStatus.active.toString(),
+        },
+      },
+    };
+  });
+
+  if (newKeyItemPutRequests.length > 0) {
+    await docClient
+      .batchWrite({
```

Comment:
nit: it seems that by using `batchWriteItems()` we'd have the table name defined once on the payload, and each item would only have into about the record to be updated - [source](https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/javascript/example_code/dynamodb/ddb_batchwriteitem.js). Defs NABD though.